x-pack/platform/plugins/shared/actions/server/lib/format_zod_error.test.ts

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ZodIssue } from '@kbn/zod';
+import { ZodError } from '@kbn/zod';
+import { formatZodError } from './format_zod_error';
+
+describe('formatZodError', () => {
+  it('formats a zod error with a single issue', () => {
+    const issue: ZodIssue = {
+      code: 'invalid_type',
+      expected: 'string',
+      received: 'boolean',
+      path: ['routingKey'],
+      message: 'Expected string, received boolean',
+    };
+    expect(formatZodError(new ZodError([issue]))).toMatchInlineSnapshot(
+      `"Field \\"routingKey\\": Expected string, received boolean"`
+    );
+  });
+
+  it('formats a zod error with a multiple issues', () => {
+    const issues: ZodIssue[] = [
+      {
+        code: 'invalid_type',
+        expected: 'string',
+        received: 'boolean',
+        path: ['routingKey'],
+        message: 'Expected string, received boolean',
+      },
+      {
+        code: 'unrecognized_keys',
+        keys: ['extraKey'],
+        path: [],
+        message: `Unrecognized key(s) in object: 'extraKey'`,
+      },
+    ];
+    expect(formatZodError(new ZodError(issues))).toMatchInlineSnapshot(`
+      "2 errors:
```

Comment:
Did we have multi-line error messages for these before zod?  Wondering what these look like from Kibana logs in Discover, and logged to console, in the UX, etc.